assembl/static2/js/app/components/debate/common/postsFilter/menu.jsx

```diff
@@ -0,0 +1,200 @@
+// @flow
+import * as React from 'react';
+import { DropdownButton } from 'react-bootstrap';
+import debounce from 'lodash/debounce';
+import classNames from 'classnames';
+import { I18n } from 'react-redux-i18n';
+import { connect } from 'react-redux';
+
+import PostsFilterMenuItem from './menuItem';
+import PostsFilterButton from './button';
+import PostsFilterButtons from './buttons';
+import PostsFilterLabelMenuItem from './label';
+import { withScreenHeight } from '../../../common/screenDimensions';
+import { setThreadPostsFilterPolicies } from '../../../../actions/threadFilterActions';
+import {
+  defaultDisplayPolicy,
+  defaultOrderPolicy,
+  defaultPostsFiltersStatus,
+  postsDisplayPolicies,
+  postsFiltersPolicies,
+  postsOrderPolicies
+} from './policies';
+
+type Props = {
+  postsDisplayPolicy: PostsDisplayPolicy,
+  postsFiltersStatus: PostsFiltersStatus,
+  postsOrderPolicy: PostsOrderPolicy,
+  screenHeight: number,
+  setPostsFilterPolicies: (
+    postsDisplay: PostsDisplayPolicy,
+    postsOrder: PostsOrderPolicy,
+    postsFiltersStatus: PostsFiltersStatus
+  ) => void
+};
+
+type State = {
+  selectedPostsDisplayPolicy: PostsDisplayPolicy,
+  selectedPostsFiltersStatus: PostsFiltersStatus,
+  selectedPostsOrderPolicy: PostsOrderPolicy,
+  sticky: boolean
+};
+
+export class DumbPostsFilterMenu extends React.Component<Props, State> {
+  componentWillMount(): void {
+    const { postsDisplayPolicy, postsOrderPolicy, postsFiltersStatus } = this.props;
+    this.setState({
+      sticky: true,
+      selectedPostsFiltersStatus: { ...postsFiltersStatus },
+      selectedPostsDisplayPolicy: postsDisplayPolicy,
+      selectedPostsOrderPolicy: postsOrderPolicy
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.setButtonPosition);
+  }
+
+  componentWillUnmount = () => {
+    window.removeEventListener('scroll', this.setButtonPosition);
+  };
+
+  setButtonPosition = debounce(() => {
+    const { screenHeight } = this.props;
+    if (window.pageYOffset > screenHeight - 60) {
+      // Show the button when we scrolled minimum the height of the window.
+      this.setState({ sticky: true });
+    } else {
+      this.setState({ sticky: false });
+    }
+  }, 100);
+
+  selectPostsDisplayPolicy = (postsDisplayPolicy: any) => {
+    // FIXME PostsDisplayPolicy
+    this.setState({ selectedPostsDisplayPolicy: postsDisplayPolicy });
+  };
+
+  selectPostsOrderPolicy = (postsOrderPolicy: any) => {
+    // FIXME PostsOrderPolicy
+    this.setState({ selectedPostsOrderPolicy: postsOrderPolicy });
+  };
+
+  selectPostsFilter = (postsFilterPolicy: any, selected?: boolean) => {
```

Comment:
juste une question, pas une remarque, pourquoi postsDisplayPolicy, postsOrderPolicy, postsFilterPolicy sont en any ?

Reply:
bonne question. en fait je suis contrarié par ça, mais on est coincés en flow. on a besoin de union types pour pouvoir utiliser les policy avec le composant PostsFilterMenuItem (qui attend un PostsFilterItem
or, en Flow on ne peut pas faire de Union types avec des objets, ça n'est permis qu'avec des types scalaires

un exemple simple :

```
interface Toto {
test: null | { subtest: string }
}

const toto: Toto = {
test: null,
}
```
```
Cannot assign object literal to toto because null [1] is incompatible with object type [2] in
property test.
```